snuba/datasets/transactions_processor.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime
+
+import ipaddress
+import uuid
+
+from snuba.processor import (
+    _as_dict_safe,
+    MessageProcessor,
+    _ensure_valid_date,
+    _unicodify
+)
+from snuba.datasets.events_processor import (
+    enforce_retention,
+    extract_base,
+    extract_extra_contexts,
+    extract_extra_tags,
+    extract_user,
+)
+
+
+class TransactionsMessageProcessor(MessageProcessor):
+    PROMOTED_TAGS = [
+        "environment",
+        "sentry:release",
+        "sentry:user",
+        "sentry:dist",
+    ]
+
+    def __extract_timestamp(self, field):
+        timestamp = _ensure_valid_date(datetime.fromtimestamp(field))
```

Comment:
This can return `None` (but shouldn't under normal circumstances) which will cause an `AttributeError` on the next line.

I don't know that the way the events processor handles it is much better, but it's at least protected here: https://github.com/getsentry/snuba/blob/b590a528d6ab90504e93678a5693b3bc899bb872/snuba/datasets/events_processor.py#L138-L140

Reply:
fixed